test_tableslurp.sh

```diff
@@ -128,27 +128,28 @@ echo
 # INDEX_FILE is:    index.txt
 # INDEX_BUCKET is:  kimlew-tablesnap-tests
 # BACKUP_BUCKET is: kimlew-tablesnap-tests
-./${PYTHON_ENV}/bin/python ./tableslurp -k "$AWS_ACC_KEY_ID" -s "$AWS_SEC_ACC_KEY" --name "${CONTAINER}" "${INDEX_BUCKET}" "${INDEX_FILE}" "${BACKUP_BUCKET}"
-echo
 echo "Running tableslurp..."
+./${PYTHON_ENV}/bin/python ./tableslurp -k "$AWS_ACC_KEY_ID" -s "$AWS_SEC_ACC_KEY" --name "${CONTAINER}" "${INDEX_BUCKET}" "${INDEX_FILE}" "${BACKUP_BUCKET}"
 echo
 
 # Check if test_aws_s3_cp.txt was successfully downloaded locally to
-# /tmp/tablesnap-test. Note: if ! diff - means there IS a difference between
-# the 2 files, i.e., returns a status code of 1, & there IS an ERROR. If
-# ! diff returns a status code of 0, there is NO difference between the files,
-# i.e., the contents match between uploaded file & downloaded file - so SUCCESS.
-if ! diff "${SRC_LOCAL_UPDOWN_FILE}" "${SRC_LOCAL_ORIG_TEST_FILE}"; then
-  echo "ERROR: ${UPDOWN_FILE} & ${ORIG_TEST_FILE} have different contents"
-  echo "UpDown file contents: "
-  cat "${SRC_LOCAL_UPDOWN_FILE}"
-  echo "Orig file contents: "
-  cat "${SRC_LOCAL_ORIG_TEST_FILE}"
-  die "${FAILURE} download ${UPDOWN_FILE} from S3"
-else
-  echo "${SUCCESS} downloaded ${UPDOWN_FILE} from S3"
-  ls -lah "${DIR_LOCAL}"
-fi
+# /tmp/tablesnap-test. If the 2 files have the same content, the download is a
+# SUCCESS, i.e., the contents match between uploaded file & downloaded file.
+diff "${SRC_LOCAL_UPDOWN_FILE}" "${SRC_LOCAL_ORIG_TEST_FILE}" > /dev/null
+case "$?" in
+  0)
+    echo "${SUCCESS} downloaded ${UPDOWN_FILE} from S3"
+    ls -lah "${DIR_LOCAL}"
+    ;;
+  1)
+    echo "ERROR: ${UPDOWN_FILE} & ${ORIG_TEST_FILE} have different contents"
+    echo "UpDown file contents: "
+    cat "${SRC_LOCAL_UPDOWN_FILE}"
+    echo "Orig file contents: "
+    cat "${SRC_LOCAL_ORIG_TEST_FILE}"
+    die "${FAILURE} download ${UPDOWN_FILE} from S3"
+    ;;
```

Comment:
`diff` can very likely return other non-zero status codes to indicate various errors. It's a good idea to add a catch-all at the end, just before the `esac` where you exit with a message like

```
  *)
    die "diff returned status ${?}"
    ;;
```

Reply:
Whoops. I forgot about the catch-all. Thanks.

Reply:
Looks good.